### ISSUE TITLE
Pipeline hanging during rna.align; waiting for return code

### ISSUE THREAD

**Issue by ghost** [ENCODE-DCC/rna-seq-pipeline#91]
Hi Otto,

Sorry to trouble you with this. How long should it take to progress beyond this point for the little chromosome 19 example that ships with the pipeline? My cluster last issued output of any kind about 9 hours ago. Any ideas as to why this might be?

```
caper run /data/project/weaverlab/LavaRuins/Code/environments/rna-seq-pipeline/rna-seq-pipeline.wdl -i /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/metadata/rna.pe.kallisto.input.json --singularity
2021-12-15 02:02:35,982|caper.cli|INFO| Cromwell stdout: /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/cromwell.out
2021-12-15 02:02:36,002|caper.caper_base|INFO| Creating a timestamped temporary directory. /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/.caper_tmp/rna-seq-pipeline/20211215_020236_002205
2021-12-15 02:02:36,002|caper.caper_runner|INFO| Localizing files on work_dir. /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/.caper_tmp/rna-seq-pipeline/20211215_020236_002205
2021-12-15 02:02:48,568|caper.caper_workflow_opts|INFO| Singularity image found in WDL metadata. wdl=/data/project/weaverlab/LavaRuins/Code/environments/rna-seq-pipeline/rna-seq-pipeline.wdl, s=encodedcc/rna-seq-pipeline:1.2.4
2021-12-15 02:03:01,738|caper.cromwell|INFO| Validating WDL/inputs/imports with Womtool...
2021-12-15 02:03:05,509|caper.cromwell|INFO| Passed Womtool validation.
2021-12-15 02:03:05,509|caper.caper_runner|INFO| launching run: wdl=/data/project/weaverlab/LavaRuins/Code/environments/rna-seq-pipeline/rna-seq-pipeline.wdl, inputs=/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/metadata/rna.pe.kallisto.input.json, backend_conf=/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/.caper_tmp/rna-seq-pipeline/20211215_020236_002205/backend.conf
2021-12-15 02:03:36,836|caper.cromwell_workflow_monitor|INFO| Workflow: id=37593de4-8077-422a-b9e6-ef2af2b5889c, status=Submitted
2021-12-15 02:03:36,873|caper.cromwell_workflow_monitor|INFO| Workflow: id=37593de4-8077-422a-b9e6-ef2af2b5889c, status=Running
2021-12-15 02:03:46,744|caper.cromwell_workflow_monitor|INFO| Task: id=37593de4-8077-422a-b9e6-ef2af2b5889c, task=rna.align:1, retry=0, status=Started, job_id=5280
2021-12-15 02:03:46,744|caper.cromwell_workflow_monitor|INFO| Task: id=37593de4-8077-422a-b9e6-ef2af2b5889c, task=rna.align:0, retry=0, status=Started, job_id=5308
2021-12-15 02:03:46,746|caper.cromwell_workflow_monitor|INFO| Task: id=37593de4-8077-422a-b9e6-ef2af2b5889c, task=rna.align:1, retry=0, status=WaitingForReturnCode
2021-12-15 02:03:46,746|caper.cromwell_workflow_monitor|INFO| Task: id=37593de4-8077-422a-b9e6-ef2af2b5889c, task=rna.align:0, retry=0, status=WaitingForReturnCode
```

**Comment by leepc12** [comment]
Please attach the following files for debugging:
- `~/.caper/default.conf`
- `/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/metadata/rna.pe.kallisto.input.json`
- `/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/cromwell.out`

**Comment by ghost** [comment]
[cromwell.pe.kallisto.out.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7722667/cromwell.pe.kallisto.out.txt)
[cromwell.se.out.2.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7722670/cromwell.se.out.2.txt)
[default.conf.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7723180/default.conf.txt)
[rna.se.input.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7723182/rna.se.input.txt)
[rna.pe.kallisto.index.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7723183/rna.pe.kallisto.index.txt)

Hi @leepc12 !! Thanks so much for getting back to me. I attached 2 index.json files and the accompanying cromwell.out files (one single end one paired end). They appear to advance to different stages so I hope that will be helpful... 

I have been wondering a few things that I cannot find in documentation.

1) in default.conf, do you need to comment each variable you do not use, or should I leave them blank?
2) In the index.json files, does it matter in what order the arguments are given?
3) In the index.json files for single end sequencing, I cannot figure out from the documentation if I am supposed to leave out `"rna.fastqs_R2" : [], ` or leave the line in there as an empty array?
4) If I provide an absolute path for all data and reference files, does it matter where there are stored?
5) Finally, the software versions being used are:
- Python v3.8.4
- Java v11.0.2
- Singularity v2.5.2 
- Caper v2.1.2

Thank you so much for helping me out. If I can provide anything else, please let me know. 
- Vincent

**Comment by leepc12** [comment]
Found a bunch of `file doesn't exist` errors in the cromwell.out log.
```
Could not localize /data/project/weaverlab/LavaRuins/Reference/rna-seq-pipeline/Human/GRCh38/common/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx -> /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/rna/37593de4-8077-422a-b9e6-ef2af2b5889c/call-kallisto/shard-0/inputs/1237465266/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx:
	/data/project/weaverlab/LavaRuins/Reference/rna-seq-pipeline/Human/GRCh38/common/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx doesn't exist
	File not found /data/project/weaverlab/LavaRuins/Reference/rna-seq-pipeline/Human/GRCh38/common/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx
	File not found /data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/rna/37593de4-8077-422a-b9e6-ef2af2b5889c/call-kallisto/shard-0/inputs/1237465266/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx -> /data/project/weaverlab/LavaRuins/Reference/rna-seq-pipeline/Human/GRCh38/common/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx
	File not found /data/project/weaverlab/LavaRuins/Reference/rna-seq-pipeline/Human/GRCh38/common/Homo_sapiens.GRCh38.cdna.all.chr19_ERCC_phix_k31_kallisto.idx
	at common.validation.Validation$ValidationTry$.toTry$extension1(Validation.scala:94)
	at common.validation.Validation$ValidationTry$.toTry$extension0(Validation.scala:90)
	at cromwell.backend.standard.StandardAsyncExecutionActor.instantiatedCommand(StandardAsyncExecutionActor.scala:669)
	... 35 common frames omitted
```
Please check if paths in your input JSON are correct.

**Comment by ghost** [comment]
They are definitely correct. Sorry that is the iteration before what I should have sent you. Will resend now.

**Comment by ghost** [comment]
[cromwell.pe.out.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7723643/cromwell.pe.out.txt)

**Comment by ghost** [comment]
Also, cat cromwell.se.out.2.txt | grep "localize" (which is found as a part of the error message for the pe run) returns nothing, so I think it cannot be the same error as the first PE index.json file I sent. Thanks again.

**Comment by ghost** [comment]
For the jobs submitted yesterday at 3PM your time, here is where we are at:

```
2021-12-15 18:09:48,247|caper.caper_runner|INFO| launching run: wdl=/data/project/weaverlab/LavaRuins/Code/environments/rna-seq-pipeline/rna-seq-pipeline.wdl, inputs=/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/metadata/rna.pe.kallisto.input.json, backend_conf=/data/project/weaverlab/LavaRuins/Data/Projects/pe_default_example/.caper_tmp/rna-seq-pipeline/20211215_180930_407506/backend.conf
2021-12-15 18:09:59,036|caper.cromwell_workflow_monitor|INFO| Workflow: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, status=Submitted
2021-12-15 18:09:59,068|caper.cromwell_workflow_monitor|INFO| Workflow: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, status=Running
2021-12-15 18:10:08,934|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.align:0, retry=0, status=Started, job_id=269302
2021-12-15 18:10:08,934|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.align:1, retry=0, status=Started, job_id=269340
2021-12-15 18:10:08,937|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.align:1, retry=0, status=WaitingForReturnCode
2021-12-15 18:10:08,937|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.align:0, retry=0, status=WaitingForReturnCode
2021-12-15 18:10:13,935|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.kallisto:1, retry=0, status=Started, job_id=269423
2021-12-15 18:10:13,935|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.kallisto:0, retry=0, status=Started, job_id=269461
2021-12-15 18:10:13,935|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.kallisto:0, retry=0, status=WaitingForReturnCode
2021-12-15 18:10:13,936|caper.cromwell_workflow_monitor|INFO| Task: id=3c4bad72-f442-4124-b08d-5bb6ce4cacae, task=rna.kallisto:1, retry=0, status=WaitingForReturnCode
```

again in this case nothing about missing files. ty for your time.

**Comment by ghost** [comment]
[cromwell.12.16.out.txt](https://github.com/ENCODE-DCC/rna-seq-pipeline/files/7731758/cromwell.12.16.out.txt)

here is an additional cromwell.out.  it has been running ~24h now on the chr 19 example with no changes

**Comment by rfran010** [comment]
Not sure if a solution was ever found, but I just tried setting up and had a very similar issue where it would just run alignment indefinitely.

My issue I found looking into stderr logs (e.g. ./rna/???/execution/shard-0/call-align/stderr)  and saw this error:
```
        INFO:    Converting OCI blobs to SIF format
        FATAL:   Unable to handle docker://encodedcc/rna-seq-pipeline:1.2.4 uri: while building SIF from layers: unable to create new build: while searching for mksquashfs: exec: "mksquashfs": executable file not found in $PATH
```
There were no errors or anything similar in the main logs, e.g. cromwell.out
The solution was to add my sbin folder to the path, and I was able to run it successfully.
This was on an hpc with slurm.
